executor/src/witgen/machines/block_machine.rs

```diff
@@ -208,6 +205,19 @@ impl<'a, T: FieldElement> ChangeLogger<'a, T> {
         }
     }
 
+    fn log_constraints(
+        &mut self,
+        bm: &BlockMachine<T>,
+        constraints: &Constraints<&PolynomialReference, T>,
+        row: Option<u64>,
+    ) {
+        for (poly, constraint) in constraints {
```

Comment:
Can you exit early in case we are not logging? Or check if it makes a difference WRT performance.